src/OpenTelemetry.AutoInstrumentation/Configuration/CompositeConfigurationSource.cs

```diff
@@ -61,7 +61,7 @@ public void Insert(int index, IConfigurationSource item)
     public string GetString(string key)
     {
         return _sources.Select(source => source.GetString(key))
-            .FirstOrDefault(value => value != null);
+             .FirstOrDefault(value => !string.IsNullOrEmpty(value));
```

Comment:
It could be enough to implement it only here if we refactor it. I can try doing it later.

Basically the idea is to change `CompositeConfigurationSource` to e.g. `Configuration`. It will not implement `IConfigurationSource` and existing code would depend on `Configuration` instead of `IConfigurationSource`. Then we will be sure that the "sanitization" is always applied.